### PR TITLE
Adding dcos-diagnostics Windows build.

### PR DESCRIPTION
**Pull request by DominikDary**
## High-level description

Adding dcos-diagnostics build.

## Corresponding DC/OS tickets (required)

  - [DCOS_OSS-5543](https://jira.mesosphere.com/browse/DCOS_OSS-5543) Adding dcos-diagnostics Windows build.

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain.
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]

### CODE REVIEW

**Comment by DominikDary** [comment]
@mesosphere-mergebot request-review  janisz

**Comment by DominikDary** [comment]
@mesosphere-mergebot request-review jongiddy

**Comment by DominikDary** [comment]
@mesosphere-mergebot changelog-not-required This is adding only a windows build and the feature is not yet available to end customers. waiting for winpanda.

File: packages/dcos-diagnostics/windows.build.ps1 [hunk 1]
```diff
@@ -0,0 +1,14 @@
+$SRC_DIR = "c:\gopath\src\github.com\dcos\dcos-diagnostics\"
+new-item -itemtype directory "c:\gopath\src\github.com\dcos"
+
+copy-item -recurse  -Path c:/pkg/src/dcos-diagnostics/ -Destination c:/gopath/src/github.com/dcos/
+
+Push-Location $SRC_DIR
+
+$env:GOOS = "windows"
```
Comment (janisz) [inline packages/dcos-diagnostics/windows.build.ps1]: Do we need to set `GOOS` when building on Windows?

File: packages/dcos-diagnostics/windows.buildinfo.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "single_source" : {
+    "kind": "git",
+    "git": "https://github.com/dcos/dcos-diagnostics.git",
+    "ref": "1b3be79e72185db0c9786d0d994984eb5d1efc4b",
```
Comment (janisz) [inline packages/dcos-diagnostics/windows.buildinfo.json]: This line duplicates https://github.com/dcos/dcos/blob/e2e77eebe36405e60e3eb1c309d0b86925bef7af/packages/dcos-diagnostics/buildinfo.json#L6 and makes it harder to bump dcos-diagnostics in the future. We should find a way to unify it.

File: pkgpanda/docker.windows/dcos-builder/Dockerfile [hunk 3]
```diff
@@ -1,63 +1,63 @@
...
-    --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 `
-    --add Microsoft.VisualStudio.Component.VC.140 `
-    --add Microsoft.VisualStudio.Component.Windows10SDK.16299.Desktop `
-    --add Microsoft.VisualStudio.Component.Windows81SDK `
-    --add Microsoft.VisualStudio.Component.VC.ATL `
-    || IF "%ERRORLEVEL%"=="3010" EXIT 0
-
-# Start developer command prompt with any other commands specified.
-
+# escape=`
```
Comment (janisz) [inline pkgpanda/docker.windows/dcos-builder/Dockerfile]: [The wall of Pink](https://www.hanselman.com/blog/YoureJustAnotherCarriageReturnLineFeedInTheWall.aspx)

File: pkgpanda/docker.windows/dcos-builder/setup-golang.ps1 [hunk 4]
```diff
@@ -1,6 +1,6 @@
-$FILENAME_PATH = "c:\temp\go1.9.3.windows-amd64.zip"
-$INSTALL_DIR = Join-Path $env:SystemDrive ""
- 
-Write-Output "Extracting $FILENAME_PATH"
-Expand-Archive $FILENAME_PATH -destinationpath $INSTALL_DIR
-
+$FILENAME_PATH = "c:\temp\go1.12.9.windows-amd64.zip"
```
Comment (janisz) [inline pkgpanda/docker.windows/dcos-builder/setup-golang.ps1]: :+1:

File: packages/dcos-diagnostics/windows.build.ps1 [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+& go build .
```
Comment (janisz) [inline packages/dcos-diagnostics/windows.build.ps1]: In Linux Makefile we also set Version and Commit, but we've never bump version so I'm fine with setting only commit.

https://github.com/dcos/dcos-diagnostics/blob/1389e1499b0790a9bdf041514fe173a395a4b7b8/Makefile#L3-L5

**Comment by janisz** [comment]
@mesosphere-mergebot request-review @drewvanstone @br-lewis @makkes @bamarni @rgo3

File: packages/dcos-diagnostics/windows.build.ps1 (hunk 1 above)
Comment (DominikDary) [reply packages/dcos-diagnostics/windows.build.ps1]: yes, when this was missing it was not building

File: packages/dcos-diagnostics/windows.buildinfo.json (hunk 2 above)
Comment (DominikDary) [reply packages/dcos-diagnostics/windows.buildinfo.json]: I don't think that is possible. Please look [here](https://github.com/dcos/dcos/tree/master/packages/mesos), mesos has as well both files.

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3  Can you review, please?

**Comment by janisz** [comment]
:wave: @drewvanstone @br-lewis @makkes @bamarni @rgo3  Can you review, please?

**Pull request merged**